server/src/main/java/org/elasticsearch/transport/TransportKeepAlive.java

```diff
@@ -52,28 +49,17 @@ final class TransportKeepAlive implements Closeable {
 
     static final int PING_DATA_SIZE = -1;
 
-    private static final BytesReference PING_MESSAGE;
-
-    static {
-        try (BytesStreamOutput out = new BytesStreamOutput()) {
-            out.writeByte((byte) 'E');
-            out.writeByte((byte) 'S');
-            out.writeInt(PING_DATA_SIZE);
-            PING_MESSAGE = out.copyBytes();
-        } catch (IOException e) {
-            throw new AssertionError(e.getMessage(), e); // won't happen
-        }
-    }
+    private static final byte[] PING_MESSAGE = {'E', 'S', -1, -1, -1, -1};
```

Comment:
Why do we need 4 `-1` here?

Reply:
It's the byte representation of the `-1` integer. 

```
ByteBuf buffer = Unpooled.buffer();
byte[] bytes = new byte[4];
buffer.writeInt(-1);
buffer.getBytes(0, bytes);
```

```
bytes: byte[4]@43
  0: -1
  1: -1
  2: -1
  3: -1
```

Reply:
:man_facepalming: of course, sorry!